### PR TITLE
adding crawling tweet

### PR DESCRIPTION
**Pull request by SCcagg5**
adding:
tweet crawler (get tweets about covid from the last 24H get username, date, content, id)
tweet crawler docker-compose

modifying:
python docker updated to python 3.8

### CODE REVIEW

File: flow/scripts/crawl_tweet.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        if 'acknowledged' in response:
+            if response['acknowledged'] == True:
+                logger.info("INDEX MAPPING SUCCESS FOR INDEX: {}".format(response['index']))
+            elif 'error' in response:
+                logger.error(response['error']['root_cause'])
+                logger.error("Error type: {}".format(response['error']['type']))
+                raise Exception("Unable to create index mapping")
+
+with Flow("Search tweets and insert") as flow:
+    bucket="tweets"
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py:113]: `bucket` is only used in S3, you might want to replace it by `index_name`

File: flow/scripts/crawl_tweet.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    flow.set_dependencies(
+        task=GetTweets(),
+        upstream_tasks=[GenerateEsMapping(bucket)],
+        keyword_tasks=dict(bucket_name=bucket))
+
+try:
+    client = Client()
+    client.create_project(project_name="pandemic-knowledge")
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py:121]: Give a unique `project_name` such as `pandemic-knowledge-crawl-tweets`

File: flow/scripts/crawl_tweet.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+except prefect.utilities.exceptions.ClientError as e:
+    logger.info("Project already exists")
+
+flow.register(project_name="pandemic-knowledge", labels=["development"])
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py:125]: Give a unique `project_name` such as `pandemic-knowledge-crawl-tweets`

File: flow/scripts/crawl_tweet.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                raise Exception("Unable to create index mapping")
+
+with Flow("Search tweets and insert") as flow:
+    bucket="tweets"
+    flow.set_dependencies(
+        task=GetTweets(),
+        upstream_tasks=[GenerateEsMapping(bucket)],
+        keyword_tasks=dict(bucket_name=bucket))
+
+try:
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py:119]: Place the code below inside a `if __name__ == "__main__":` statement (see https://github.com/flavienbwk/Pandemic-Knowledge/blob/main/flow/scripts/crawl_google_news.py)

File: flow/scripts/crawl_tweet.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+    helpers.bulk(es_inst, actions)
+
+class GetTweets(Task):
+    def run(self, bucket_name):
+        d = datetime.now() - timedelta(days=1)
+        limit = d.strftime("%y-%m-%d")
+        to_inject = []
+        for i,tweet in enumerate(sntwitter.TwitterSearchScraper("covid since:" + limit).get_items()):
+            if i>300:
```
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py:74]: Make this limit a global variable at top of script

(maybe we should lower it to 100 for example too)

File: flow/scripts/crawl_tweet.py (hunk 1 above)
Comment (flavienbwk) [inline flow/scripts/crawl_tweet.py:113]: Set `index_name` as a global variable at top of script

**Review by flavienbwk — CHANGES_REQUESTED** [review]

**Comment by SCcagg5** [comment]
merge https://github.com/flavienbwk/Pandemic-Knowledge/pull/8

**Pull request closed without merging**